internal/http/middleware/auth.go

```diff
@@ -14,13 +16,35 @@ import (
 // request context.
 type AuthMiddleware struct {
 	deps model.Dependencies
+
+	trustedIPs []*net.IPNet
 }
 
 func NewAuthMiddleware(deps model.Dependencies) *AuthMiddleware {
-	return &AuthMiddleware{deps: deps}
+	plainIPs := deps.Config().Http.SSOTrustedProxy
```

Comment:
Can we move the SSO logic to a new middleware? That way we would only inject it if needed (`SSOEnabled == true`)

Having the logic split would prove useful and is better for long term maintenance. You could even put it in a different file (`auth_sso_proxy.go`?)